CMakeLists.txt

```diff
@@ -167,7 +175,11 @@ else()
   else()
     add_nondebug_compile_flag("-O2")
   endif()
-  add_nondebug_compile_flag("-UNDEBUG") # Keep asserts.
+  if(NOT BYN_ENABLE_ASSERTIONS)
```

Comment:
isn't the condition here, and above, flipped? If `BYN_ENABLE_ASSERTIONS`,  then we need to enter the if?

Reply:
Might be good to add a unit test for assertions working (which would only work if the build actually was built that way).

Reply:
Ah, I thanks.  I got confused while testing.  Fixed now.

In terms of a test, I think I'll just error out at  compile time for now.